### PR TITLE
Apply upstream (rest-server) API changes

### PR DESCRIPTION
**Pull request by claui**
In June 2018, dependency [rest-server](https://github.com/restic/rest-server) cleaned up [its API a little](https://github.com/restic/rest-server/pull/67), breaking the restic Caddy plugin build in the process.

Since then, attempting to build the restic Caddy plugin produces the following errors:

```
build/src/github.com/restic/caddy/setup.go:24:4: undefined: restserver.Config
build/src/github.com/restic/caddy/setup.go:40:19: undefined: restserver.NewMux
```

This PR applies the upstream API changes to fix the errors.

**Note:** This is a best-effort fix. I don’t even use restic, and saw the build error by pure coincidence because I happened to include the plugin accidentally. This fix is therefore completely untested.
**If possible, please test before merging.**

### CODE REVIEW

File: setup.go [hunk 1]
```diff
@@ -15,13 +15,17 @@ func init() {
 
 func setup(c *caddy.Controller) error {
 	var basePath string
+	var restConfig restserver.Server
+	var httpConfig *httpserver.SiteConfig
```
Comment (mholt) [inline setup.go]: Get rid of these zero-value declarations; variables are both assigned below

File: setup.go [hunk 2]
```diff
@@ -15,13 +15,17 @@ func init() {
...
+
+	restConfig = restserver.Server{}
```
Comment (mholt) [inline setup.go]: Turn this into a `:=`

File: setup.go [hunk 3]
```diff
@@ -32,15 +36,15 @@ func setup(c *caddy.Controller) error {
 		basePath = "/"
 	}
 
-	cfg := httpserver.GetConfig(c)
+	httpConfig = httpserver.GetConfig(c)
```
Comment (mholt) [inline setup.go]: Turn this into a `:=`

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks! I've been meaning to get around to this. I think this is pretty close, but I'll need to test it out like you said. Just a couple minor changes in the meantime.

**Comment by claui** [comment]
Done.

**Comment by claui** [comment]
@mholt Have you had a chance to test the changes yet?

**Review by mholt — APPROVED** [review]

**Comment by mholt** [comment]
Thanks for the reminder! Sorry I lost track of this. Appreciate your contribution, especially since you're not even a restic user. :)

**Pull request merged**